src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -223,7 +223,11 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
   }
 
   public supportsApiType(api: ApiType): boolean {
-    return api === ApiType.WMS || (api === ApiType.PROCESSING && !!this.dataset);
+    if (this.dataProduct) {
+      return api === ApiType.WMS;
+    } else {
+      return api === ApiType.WMS || (api === ApiType.PROCESSING && !!this.dataset);
+    }
```

Comment:
Just a minor nitpick... It is usually easier to follow code from top to bottom, than it is to understand the nesting levels. Because of that I would suggest removing unneeded `else` block:
```
    if (this.dataProduct) {
      return api === ApiType.WMS;
    };
    return api === ApiType.WMS || (api === ApiType.PROCESSING && !!this.dataset);
```

Fftr.